src/config.ts

```diff
@@ -1,80 +1,57 @@
-import * as findUp from "find-up";
-import * as fs from "fs";
-import * as yaml from "yaml";
-import { promisify } from "util";
-import * as path from "path";
-import * as glob from "fast-glob";
-import * as signale from "signale";
-import * as jf from "joiful";
-import { env, cwd as getProcessCwd } from "process";
-import { ProjectFileException } from "./errors";
-
-const readFile = promisify(fs.readFile);
-
-const CONFIG_FILE_NAME = "docmaker.yaml";
+import * as findUp from 'find-up';
+import * as yaml from 'yaml';
+import * as path from 'path';
+import * as glob from 'fast-glob';
+import * as signale from 'signale';
+import * as jf from 'joiful';
+import {cwd as getProcessCwd, env} from 'process';
+import ProjectFileError from './errors/project-file-error';
+import {readFile} from './utils/file-utils';
+import ValidationError from './errors/validation-error';
+
+const CONFIG_FILE_NAME = 'docmaker.yaml';
 
 /**
  * The configuration for Docmaker
  */
-class Config {
-  @(jf.string().required())
+export class Config {
+
+  @jf.string().required()
   public layout!: string;
 
-  @(jf.string().default("build"))
+  @jf.string().default('build')
   public buildDir!: string;
 
-  @(jf
-    .array()
-    .default([])
-    .items(j => j.string()))
+  @jf.array().default([]).items(t => t.string())
   public pages!: string[];
 
-  @(jf
-    .array()
-    .default([])
-    .items(j => j.string()))
+  @jf.array().default([]).items(t => t.string())
   public data!: string[];
 
-  @(jf
-    .array()
-    .default([])
-    .items(j => j.string()))
+  @jf.array().default([]).items(t => t.string())
   public assets!: string[];
 
   public static async fromDirectory(projectDir: string): Promise<Config> {
-    let config = new Config();
-
     const configPath = path.join(projectDir, CONFIG_FILE_NAME);
     const content = await readFile(configPath);
-
     // Load yaml config, use "Config" instead of "any"
-    let loadedConfig: Config = yaml.parse(content.toString());
-
+    const parsedConfig: Config = yaml.parse(content.toString());
```

Comment:
When `docmaker.yaml` is empty, this will return `null`. We should probably handle that

Reply:
@bramceulemans could you handle this?

Reply:
Sure thing, let's make an issue for it and fix it seperatly.

Reply:
sure